backend/app/Services/VisitorsFlow/FlowAggregateService.php

```diff
@@ -0,0 +1,409 @@
+use App\Repositories\Elasticsearch\VisitorsFlow\Contracts\VisitorFlowScreenRepository;
+use App\Repositories\Elasticsearch\VisitorsFlow\CountryCriteria;
+use App\Repositories\Elasticsearch\VisitorsFlow\DeviceCriteria;
+use App\Repositories\Elasticsearch\VisitorsFlow\ScreenCriteria;
+use Carbon\Carbon;
+
+final class FlowAggregateService
+{
+    private $pageRepository;
+    private $visitRepository;
+    private $visitorFlowCountryRepository;
+    private $websiteRepository;
+    private $geoPositionRepository;
+    private $visitorFlowBrowserRepository;
+    private $visitorFlowDeviceRepository;
+    private $visitorFlowScreenRepository;
+
+    public function __construct(
+        PageRepository $pageRepository,
+        VisitRepository $visitRepository,
+        VisitorFlowCountryRepository $visitorFlowCountryRepository,
+        WebsiteRepository $websiteRepository,
+        GeoPositionRepository $geoPositionRepository,
+        VisitorFlowBrowserRepository $visitorFlowBrowserRepository,
+        VisitorFlowDeviceRepository $visitorFlowDeviceRepository,
+        VisitorFlowScreenRepository $visitorFlowScreenRepository
+    ) {
+        $this->pageRepository = $pageRepository;
+        $this->visitRepository = $visitRepository;
+        $this->visitorFlowCountryRepository = $visitorFlowCountryRepository;
+        $this->websiteRepository = $websiteRepository;
+        $this->geoPositionRepository = $geoPositionRepository;
+        $this->visitorFlowBrowserRepository = $visitorFlowBrowserRepository;
+        $this->visitorFlowDeviceRepository = $visitorFlowDeviceRepository;
+        $this->visitorFlowScreenRepository = $visitorFlowScreenRepository;
+    }
+
+    public function aggregate(Visit $visit): void
+    {
+        $previousVisit = $this->getLastVisit($visit);
+        $isFirstInSession = $previousVisit === null;
+        $level = $this->getLevel($visit, $isFirstInSession);
+
+        $countryAggregate = $this->getCountryAggregate($visit, $level, $isFirstInSession, $previousVisit);
+        $browserAggregate = $this->getBrowserAggregate($visit, $level, $isFirstInSession, $previousVisit);
+        $deviceAggregate = $this->getDeviceAggregate($visit, $level, $isFirstInSession, $previousVisit);
+        $screenAggregate = $this->getScreenAggregate($visit, $level, $isFirstInSession, $previousVisit);
+
+        $this->updateCountryAggregate($visit, $level, $previousVisit, $countryAggregate);
+        $this->updateBrowserAggregate($visit, $level, $previousVisit, $browserAggregate);
+        $this->updateDeviceAggregate($visit, $level, $previousVisit, $deviceAggregate);
+        $this->updateScreenAggregate($visit, $level, $previousVisit, $screenAggregate);
+    }
+
+    private function getLevel(Visit $visit, bool $isFirstInSession): int
+    {
+        if (!$isFirstInSession) {
+            return $this->getVisitsCount($visit);
+        }
+        return 1;
```

Comment:
add constant FIRST_LEVEL = 1;

and change condition

if ($isFirstInSession) {
     return self::FIRST_LEVEL;
}

return $this->getVisitsCount($visit);